cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -63,6 +63,39 @@ void executeBinaryOp(T left, T right, T output, OpFunc op) {
       });
 }
 
+template <JITTypeTag Type, typename NativeType, typename BuilderType>
+void executeCompareSingleParamTest(NativeType input, bool output, BuilderType builder) {
+  LLVMJITModule module("TestOperator");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_operator_func",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::BOOL},
+      .params_type = {JITFunctionParam{.name = "x", .type = Type}},
+  });
+  builder(function.get());
+  function->finish();
+  module.finish();
+
+  auto func_ptr = function->getFunctionPointer<bool, NativeType>();
+  EXPECT_EQ(func_ptr(input), output);
+}
+
+template <JITTypeTag Type, typename T>
+void executeCompareOp(T left, T right, bool output, OpFunc op) {
+  using NativeType = typename JITTypeTraits<Type>::NativeType;
+  executeCompareSingleParamTest<Type>(
+      static_cast<NativeType>(left),
```

Comment:
why input have only `left`, `right` should also be input?

Reply:
> why input have only `left`, `right` should also be input?

Yes, 'right' also can be input. We handled it in OpFunc to do some conversions as it could be constant, It can be understood that the input is obtained ‘left’ after the op calculation such as  'left operator right'. so input have only 'left' .